### ISSUE TITLE
Catalog item composition vs Mixins

### ISSUE THREAD

**Issue by nf-s** [TerriaJS/terriajs#4705]
I raise this in issue https://github.com/TerriaJS/terriajs/issues/4704

> A heap of layers don't support ordering which can quite easily - for example any layers which create GeoJsonCatalogItems.
This opens up a broader issue of composition of different CatalogItems, and having to connect properties (for example show) for things to work properly.

> Maybe we should create a GeoJsonMixin (and CsvMixin) so we can just shove them into a CatalogItem without having to worry about forgetting to define all of the properties.
I think this could speed up development time quite a bit and it would make simplify quite a few CatalogItems

I think I remember @steve9164 talking about how we should put as much functionality into Mixins and keep CatalogItems lean - is that right?

**Comment by KeyboardSounds** [comment]
Yep! This is exactly what Mixins are for. See: https://github.com/TerriaJS/terriajs/blob/next/lib/ModelMixins/Cesium3dTilesMixin.ts

**Comment by nf-s** [comment]
https://github.com/TerriaJS/terriajs/issues/5380

**Comment by nf-s** [comment]
Also, we should consolidate Mixins/Traits
